service/src/main/java/uk/gov/hmcts/reform/fpl/service/MessageJudgeService.java

```diff
@@ -81,6 +83,21 @@
         return data;
     }
 
+    public List<String> validateJudgeReplyMessage(CaseData caseData) {
+        List<String> errors = new ArrayList<>();
+
+        MessageJudgeEventData messageJudgeEventData = caseData.getMessageJudgeEventData();
+        JudicialMessage judicialMessageReply = messageJudgeEventData.getJudicialMessageReply();
+
+        if (judicialMessageReply != null && YES.getValue().equals(judicialMessageReply.getIsReplying())
+            && isNotEmpty(judicialMessageReply.getReplyFrom())
+            && judicialMessageReply.getReplyFrom().equals(judicialMessageReply.getReplyTo())) {
```

Comment:
minor, I wonder if we could break this up into smaller more readable private methods e.g

`if (isReplyingToMessge(judicialMessageReply) && hasMatchingReplyEmaiAddress(judicialMessageReply))`